Proper ordering of the handlers chapters and addition of some examples in the Logging guide (#34860)
quarkusio/quarkus: docs/src/main/asciidoc/logging.adoc

* A per-category configuration example:
+
[source, properties]
----
quarkus.log.handler.file.my-file-handler.enable=true
quarkus.log.handler.file.my-file-handler.path=application.log
quarkus.log.handler.file.my-file-handler.format=%d{yyyy-MM-dd HH:mm:ss} [com.example] %s%e%n

quarkus.log.category."com.example".handlers=my-file-handler
quarkus.log.category."com.example".use-parent-handlers=false
----

For details about its configuration, see the xref:#quarkus-log-logging-log-config_quarkus.log.file-file-logging[file logging configuration] reference.

=== Syslog log handler

The syslog handler in Quarkus follows the link:https://en.wikipedia.org/wiki/Syslog[Syslog] protocol, which is used to send log messages on Unix-like systems.
It utilizes the protocol defined in link:https://tools.ietf.org/html/rfc5424[RFC 5424].

By default, the syslog handler is disabled.
When enabled, it sends all log events to a syslog server, typically the local syslog server for the application.

* A global configuration example:
+
[source, properties]
----
quarkus.log.syslog.enable=true
quarkus.log.syslog.app-name=my-application
quarkus.log.syslog.format=%d{yyyy-MM-dd HH:mm:ss} %-5p [%c] (%t) %s%e%n
----

> +

Reviewer: ```suggestion
```

Missed this, sorry :-)

NOTE: Printing the details can be expensive as the values are retrieved from the caller.
The details include the source class name, source file name, source method name, and source line number.

== Log handlers

A log handler is a logging component responsible for the emission of log events to a recipient.
Quarkus includes several different log handlers: **console**, **file**, and **syslog**.

> The featured examples uses `com.example` as the name of a logging category.

Reviewer: ```suggestion
The featured examples use `com.example` as a logging category.
```

----
quarkus.log.file.enable=true
quarkus.log.file.path=application.log
quarkus.log.file.format=%d{yyyy-MM-dd HH:mm:ss} %-5p [%c] (%t) %s%e%n
----

* A per-category configuration example:
+
[source, properties]
----
quarkus.log.handler.file.my-file-handler.enable=true
quarkus.log.handler.file.my-file-handler.path=application.log
quarkus.log.handler.file.my-file-handler.format=%d{yyyy-MM-dd HH:mm:ss} [com.example] %s%e%n

quarkus.log.category."com.example".handlers=my-file-handler
quarkus.log.category."com.example".use-parent-handlers=false
----

For details about its configuration, see the xref:#quarkus-log-logging-log-config_quarkus.log.file-file-logging[file logging configuration] reference.

=== Syslog log handler

The syslog handler in Quarkus follows the link:https://en.wikipedia.org/wiki/Syslog[Syslog] protocol, which is used to send log messages on Unix-like systems.
It utilizes the protocol defined in link:https://tools.ietf.org/html/rfc5424[RFC 5424].

By default, the syslog handler is disabled.
When enabled, it sends all log events to a syslog server, typically the local syslog server for the application.

* A global configuration example:
+
[source, properties]
----

> quarkus.log.syslog.handlers = com.example.GlobalSyslogHandler
> quarkus.log.syslog."com.example.GlobalSyslogHandler".level = INFO
> quarkus.log.syslog."com.example.GlobalSyslogHandler".host = syslog.example.com
> quarkus.log.syslog."com.example.GlobalSyslogHandler".port = 514

Reviewer: ```suggestion
quarkus.log.syslog.enable=true
quarkus.log.syslog.app-name=my-application
quarkus.log.syslog.format=%d{yyyy-MM-dd HH:mm:ss} %-5p [%c] (%t) %s%e%n
```

----
+

> The INFO level log records for all other categories will also be sent to the same Syslog server.
> +
> The Syslog server address and port are specified in the host and port properties.

Reviewer: ```suggestion
```

* A per-category configuration example:
+
[source, properties]
----
quarkus.log.syslog.handlers = com.example.MyCategorySyslogHandler
quarkus.log.syslog.com.example."MyCategorySyslogHandler".level = WARN
quarkus.log.syslog.com.example."MyCategorySyslogHandler".host = syslog.example.com
quarkus.log.syslog.com.example."MyCategorySyslogHandler".port = 514
----

> +
> The WARN level log records for the `com.example.MyCategory` category will be sent to the specified Syslog server.

Reviewer: ```suggestion
```

+
[source, properties]
----
quarkus.log.syslog.handlers = com.example.MyCategorySyslogHandler
quarkus.log.syslog.com.example."MyCategorySyslogHandler".level = WARN
quarkus.log.syslog.com.example."MyCategorySyslogHandler".host = syslog.example.com

> quarkus.log.syslog.com.example."MyCategorySyslogHandler".port = 514

Reviewer 2: I don't think this is correct.

Author: @dmlloyd Hello David. Could I ask you for a review and maybe better Syslog examples?

Author: @Ladicek OK, thank you somuch, Laďa, for the examples! Patched the examples and pushed the commit.
Cheers

Reviewer: Should be fixed now. I actually verified all the examples here using https://github.com/Ladicek/quarkus-tinies/tree/main/logging

    }
}
----
+
In this example, we exclude log records containing specific text from console logs.
The specific text to filter on is not hard-coded; instead, it is read from the `my-filter.part` configuration property.
+
.An example of Configuring the filter in `application.properties`:
[source, properties]
----
my-filter.part=TEST
----

. Attach the filter to the corresponding handler using the `filter` configuration property, located in `application.properties`:
+
[source, properties]
----
quarkus.log.console.filter=my-filter
----

== Logging configurations examples

This chapter provides examples of frequently used logging configurations.

> * An example of a global console handler with additional per-category log level:

Author: @dmlloyd David, here I want to ask you about these leading bulleted sentences. I kinda like the idea to introduce the example a bit, however, I am sure I did not use the proper wording. Are you for tweaking those sentences with your suggestions of for their removal?
Thank you

Reviewer: The examples are introduced by the "titles" IMHO. In this case, there's

> .Console DEBUG logging except for Quarkus logs (INFO), no color, shortened time, shortened category prefixes

which I think is perfectly enough.

Author: OK, I removed that.

quarkus.log.category."io.quarkus.category".handlers=STRUCTURED_LOGGING,STRUCTURED_LOGGING_FILE
----

> * The use of named handlers, as shown in this example, is typically per-category; however, in this case, it is demonstrated with the `root` category, acting as a global handler:

Reviewer: ```suggestion
* The use of named handlers is typically per-category; however, in this case, it is demonstrated with the root category, acting as a global handler:
```
